### PR TITLE
Answered all questions

### PR DESCRIPTION
**Pull request by evancoupal321**

### CODE REVIEW

File: Problem2/src/problem2/Problem2.java [hunk 1]
```diff
@@ -5,17 +5,65 @@
...
+       }
+    }
+    
+    /**
+     * Add the digits of an integer together
+     * 
+     * @param n The integer whose digits you are adding
+     * @return The sum of the digits
+     */
+    public static int addDigits(int n)
```
Comment (cgallinaro) [inline Problem2/src/problem2/Problem2.java:43]: Your algorithm is not recursive.

File: Problem3/test/problem3/Problem3Test.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        personList[7] = new Person("38", "person8", 16);
+        personList[8] = new Person("87", "person9", 16);
+        personList[9] = new Person("82", "person10", 16);
+    }
+
+    /**
+     * Test of insertionSort method, of class Problem3.
+     */
+    @Test
+    public void testInsertionSort()
```
Comment (cgallinaro) [inline Problem3/test/problem3/Problem3Test.java:41]: Your testing should include edge cases like empty arrays and arrays of a single element.

File: Problem4/test/problem4/Problem4Test.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        numbers[7] = 24;
+        numbers[8] = 12;
+        numbers[9] = 73;
+    }
+
+    /**
+     * Test of swapSort method, of class Problem4.
+     */
+    @Test
+    public void testSwapSort() {
```
Comment (cgallinaro) [inline Problem4/test/problem4/Problem4Test.java:41]: Your testing should include edge cases like empty arrays and arrays of a single element.

File: Problem4/test/problem4/Problem4Test.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        assertArrayEquals(expResult, result); //Checks if the result matches the manual placing
+    }
+
+    /**
+     * Test of swap method, of class Problem4.
+     */
+    @Test
+    public void testSwap() {
+        System.out.println("swap");  
+        Random rand = new Random();
```
Comment (cgallinaro) [inline Problem4/test/problem4/Problem4Test.java:68]: As a general rule, we avoid writing tests that have random behaviour (though there are some exceptions to this).  With random behaviour, you can potentially run into a case where your test becomes inconsistent (i.e. sometimes passes and sometimes fails).  It makes it difficult to isolate problems if the test happens to fail.

File: Problem5/src/problem5/Problem5.java [hunk 5]
```diff
@@ -5,17 +5,181 @@
...
+    }
+    
+    /**
+     * Compares two strings to see which comes first alphabetically
+     * 
+     * @param first The first string
+     * @param second The second string
+     * @return 0 if the strings are equal, 1 if the first string comes first, 2 if the second string comes first
+     */
+    public static int compareStrings(String first, String second)
```
Comment (cgallinaro) [inline Problem5/src/problem5/Problem5.java:150]: Since the words were all lower case, you could have used the compareTo method to determine if the input string was greater or less than the middle string.

File: Problem5/test/problem5/Problem5Test.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+            indexes[indexNum] = 3;
+            indexNum++;
+        }
+    }
+    
+    /**
+     * Test of binarySearch method, of class Problem5.
+     */
+    @Test
+    public void testBinarySearch() {
```
Comment (cgallinaro) [inline Problem5/test/problem5/Problem5Test.java:57]: Your testing should include edge cases like empty arrays and arrays of a single element.

File: Problem5/test/problem5/Problem5Test.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        assertEquals(2, result3);
+        assertEquals(3, result4);
+        assertEquals(-1, result5);
+    }
+
+    /**
+     * Test of compareStrings method, of class Problem5.
+     */
+    @Test
+    public void testCompareStrings() {
```
Comment (cgallinaro) [inline Problem5/test/problem5/Problem5Test.java:75]: Your compareStrings testing never tests the case where one word is a prefix for another word.  Your implementation of compareStrings has an if statement at the end to handle this case, but you have no test coverage for it.

**Comment by cgallinaro** [comment]
Problem 1: 5/5
Problem 2: 4/5
Problem 3: 4.5/5
Problem 4: 9.5/10
Problem 5: 9/10

Total: 32/35
